differs/rpm_diff.go

```diff
@@ -0,0 +1,312 @@
+		}
+		defer os.Remove(imageName)
+
+		imageName, err = loadArchiveToDaemon(imageName, client)
+		if err != nil {
+			return packages, fmt.Errorf("Error loading archive: %s", err)
+		}
+		logrus.Debugf("Loaded archive as image %s", imageName)
+		// find out if we have to remove the image
+		inspImg, err := client.InspectImage(imageName)
+		if err != nil {
+			return packages, fmt.Errorf("Error inspecting image: %s", err)
+		}
+		// if the image has tags, it's identical to an already loaded
+		// image and must not be removed
+		if len(inspImg.RepoTags) == 0 {
+			defer client.RemoveImage(imageName)
+		}
+	}
+
+	contConf := godocker.Config{
+		Cmd:   []string{"rpm", "-qa", "--qf", "%{NAME}\t%{VERSION}\t%{SIZE}\n"},
+		Image: imageName,
+	}
+
+	hostConf := godocker.HostConfig{
+		AutoRemove: true,
+	}
+
+	contOpts := godocker.CreateContainerOptions{Config: &contConf}
+	container, err := client.CreateContainer(contOpts)
+	if err != nil {
+		return packages, err
+	}
+	logrus.Debugf("Created container %s", container.ID)
+
+	removeOpts := godocker.RemoveContainerOptions{
+		ID: container.ID,
+	}
+	defer client.RemoveContainer(removeOpts)
+
+	if err := client.StartContainer(container.ID, &hostConf); err != nil {
+		return packages, err
+	}
+
+	exitCode, err := client.WaitContainer(container.ID)
+	if err != nil {
+		return packages, err
+	}
+
+	outBuf := new(bytes.Buffer)
+	errBuf := new(bytes.Buffer)
+	logOpts := godocker.LogsOptions{
+		Context:      context.Background(),
+		Container:    container.ID,
+		Stdout:       true,
+		Stderr:       true,
+		OutputStream: outBuf,
+		ErrorStream:  errBuf,
+	}
+
+	if err := client.Logs(logOpts); err != nil {
+		return packages, err
+	}
+
+	if exitCode != 0 {
+		return packages, fmt.Errorf("non-zero exit code %d: %s", exitCode, errBuf.String())
+	}
+
+	output := strings.Split(outBuf.String(), "\n")
+	return parsePackageData(output)
+}
+
+// parsePackageData parses the package data of each line in rpmOutput and
+// returns a map of packages.
+func parsePackageData(rpmOutput []string) (map[string]util.PackageInfo, error) {
+	packages := make(map[string]util.PackageInfo)
+
+	for _, output := range rpmOutput {
+		spl := strings.Split(output, "\t")
+		if len(spl) != 3 {
+			// ignore the empty (last) line
+			if output != "" {
+				logrus.Errorf("unexpected rpm-query output: '%s'", output)
+			}
+			continue
+		}
+		pkg := util.PackageInfo{}
+
+		var err error
+		pkg.Size, err = strconv.ParseInt(spl[2], 10, 64)
+		if err != nil {
+			return packages, fmt.Errorf("error converting package size: %s", spl[2])
+		}
+
+		pkg.Version = spl[1]
+		packages[spl[0]] = pkg
+	}
+
+	return packages, nil
+}
+
+// loadArchiveToDaemon loads the image specified by archive to the docker
+// Daemon.
+func loadArchiveToDaemon(archive string, client *godocker.Client) (string, error) {
+	// read and write tar archive to input stream/buffer
+	inBuf := new(bytes.Buffer)
+	if content, err := ioutil.ReadFile(archive); err != nil {
```

Comment:
nit: I think you can avoid creating the inBuf buffer and reading the entire archive into memory by passing an open File to LoadImageOptions: https://golang.org/pkg/os/#Open

Reply:
I will check that, thanks!